### PR TITLE
user-guide.md: add a few examples and remove mentions about futures

### PR DESCRIPTION
**Pull request by kingslef**
Add a few examples to the user guide docs and remove the mentions about futures so that it is a bit closer to what's actually in finchx.

Also add a link to giter8 template to the index.

### CODE REVIEW

File: docs/mdoc/index.md [hunk 1]
```diff
@@ -53,6 +53,8 @@ object Main extends App {
 }
 ```
 
+Try also [our giter8 template](https://github.com/finch/finch.g8) to get started: `sbt new finch/finch.g8`.
```
Comment (sergeykolbasov) [inline docs/mdoc/index.md:56]: that one requires some update before being mentioned in a doc

File: docs/mdoc/user-guide.md [hunk 2]
```diff
@@ -756,7 +776,7 @@ or `beShorterThan(n: Int)`.
 
 ### Errors
 
-An endpoint may fail (it may evaluate into a `Future.exception`) by a number of reasons: it was
+An endpoint may fail (it may evaluate into a `Throwable`) by a number of reasons: it was
```
Comment (sergeykolbasov) [inline docs/mdoc/user-guide.md:779]: I think it might evaluate into `F[Throwable]`

File: docs/mdoc/index.md (hunk 1 above)
Comment (kingslef) [inline docs/mdoc/index.md:56]: I didn't realize it was quite old. Perhaps merge https://github.com/finch/finch.g8/pull/2 to get it into a better shape?

Comment (kingslef) [reply docs/mdoc/index.md:56]: I'll drop it, but I think it would be nice to have it here once it is updated.

File: docs/mdoc/user-guide.md (hunk 2 above)
Comment (kingslef) [reply docs/mdoc/user-guide.md]: fixed.

**Pull request merged**